flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -414,6 +448,18 @@ private void resumeContinuousRefreshJob(
                 deserializeContinuousHandler(
                         catalogMaterializedTable.getSerializedRefreshHandler());
 
+        // Repeated resume continuous refresh job is not supported
+        if (CatalogMaterializedTable.RefreshStatus.ACTIVATED
+                == catalogMaterializedTable.getRefreshStatus()) {
+            JobStatus jobStatus = getJobStatus(operationExecutor, handle, refreshHandler);
+            if (isJobActivated(jobStatus)) {
```

Comment:
Can we use !jobStatus.isTerminalState() here ?

Reply:
Good catch, I think we can.